firmware/screen/src/ui/screen_launcher.cpp

```
@@ -0,0 +1,338 @@
+                               // i.e. APP_BOOTING and APP_DISCONNECTED both
+                               // show it; APP_CONNECTED / APP_LAUNCHER_IDLE
+                               // hide it. (sourcery clarification on PR #91.)
+    lv_timer_t* poll_timer;    // 500 ms tick to refresh the pill
+    app_state_t last_state;    // last value seen by the timer
+    bool        first_render;  // true until the first apply_pill_state() call
+};
+
+// --- Helpers ---------------------------------------------------------------
+
+// Apply the connected/disconnected look to the status pill widgets.
+void apply_pill_state(launcher_ctx_t* ctx, app_state_t s) {
```

Clang (14.0.6)

[warning] parameter name 's' is too short, expected at least 3 characters

(readability-identifier-length)